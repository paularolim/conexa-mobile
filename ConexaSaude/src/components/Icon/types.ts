export type IconName = 'calendar' | 'house' | 'plus' | 'user' | 'eye-on' | 'eye-off';

export interface IconProps {
  name: IconName;
  width?: number;
  height?: number;
  fill?: string;
}
