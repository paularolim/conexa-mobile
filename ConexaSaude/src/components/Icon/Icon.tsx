import Calendar from '@assets/icons/calendar.svg';
import EyeOff from '@assets/icons/eye-off.svg';
import EyeOn from '@assets/icons/eye-on.svg';
import House from '@assets/icons/house.svg';
import Plus from '@assets/icons/plus.svg';
import SignOut from '@assets/icons/sign-out.svg';
import User from '@assets/icons/user.svg';

import { IconProps } from './types';

export function Icon({ name, width = 40, height = 40, fill = 'none' }: IconProps) {
  switch (name) {
    case 'calendar':
      return <Calendar width={width} height={height} stroke={fill} />;
    case 'house':
      return <House width={width} height={height} stroke={fill} />;
    case 'plus':
      return <Plus width={width} height={height} fill={fill} />;
    case 'user':
      return <User width={width} height={height} stroke={fill} />;
    case 'eye-off':
      return <EyeOff width={width} height={height} stroke={fill} />;
    case 'eye-on':
      return <EyeOn width={width} height={height} stroke={fill} />;
    case 'sign-out':
      return <SignOut width={width} height={height} stroke={fill} />;
    default:
      throw new Error('Icon name is required');
  }
}
