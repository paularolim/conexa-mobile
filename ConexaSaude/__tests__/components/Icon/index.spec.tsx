import renderer from 'react-test-renderer';

import { Icon } from '@components/Icon';

import { Wrapper } from '../../Wrapper';

describe('Icon component', () => {
  it('should render Calendar component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Icon name="calendar" />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render EyeOff component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Icon name="eye-off" />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render EyeOn component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Icon name="eye-on" />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render House component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Icon name="house" />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Plus component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Icon name="plus" />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render User component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Icon name="user" />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render SignOut component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Icon name="sign-out" />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
