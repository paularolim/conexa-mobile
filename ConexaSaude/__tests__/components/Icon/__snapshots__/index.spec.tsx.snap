// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should render Calendar component correctly 1`] = `
<svg
  data-file-name="SvgCalendar"
  height={40}
  stroke="none"
  width={40}
/>
`;

exports[`Icon component should render EyeOff component correctly 1`] = `
<svg
  data-file-name="SvgEyeOff"
  height={40}
  stroke="none"
  width={40}
/>
`;

exports[`Icon component should render EyeOn component correctly 1`] = `
<svg
  data-file-name="SvgEyeOn"
  height={40}
  stroke="none"
  width={40}
/>
`;

exports[`Icon component should render House component correctly 1`] = `
<svg
  data-file-name="SvgHouse"
  height={40}
  stroke="none"
  width={40}
/>
`;

exports[`Icon component should render Plus component correctly 1`] = `
<svg
  data-file-name="SvgPlus"
  fill="none"
  height={40}
  width={40}
/>
`;

exports[`Icon component should render User component correctly 1`] = `
<svg
  data-file-name="SvgUser"
  height={40}
  stroke="none"
  width={40}
/>
`;
